src/ConfigCatClient.ts

```diff
@@ -240,188 +248,212 @@ export class ConfigCatClient implements IConfigCatClient {
-                            return;
+                            return new SettingKeyValue(settingKey, rolloutRule.value);
                         }
                     }
                 }
 
                 const percentageItems = settings[settingKey].rolloutPercentageItems;
                 if (percentageItems && percentageItems.length > 0) {
-                    for (let i: number = 0; i < percentageItems.length; i++) {
+                    for (let i = 0; i < percentageItems.length; i++) {
                         const percentageItem: RolloutPercentageItem = percentageItems[i];
                         if (variationId === percentageItem.variationId) {
-                            resolve({ settingKey: settingKey, settingValue: percentageItem.value });
-                            return;
+                            return new SettingKeyValue(settingKey,  percentageItem.value);
                         }
                     }
                 }
             }
 
             this.options.logger.error("Could not find the setting for the given variation ID: " + variationId);
-            resolve(null);
-        });
+        }
+        catch (err) {
+            this.options.logger.error("Error occurred in getKeyAndValueAsync().\n" + errorToString(err, true));
+        }
+
+        return null;
     }
 
     getAllValues(callback: (result: SettingKeyValue[]) => void, user?: User): void {
         this.options.logger.debug("getAllValues() called.");
-        this.getAllValuesAsync(user).then(value => {
-            callback(value);
-        });
+        this.getAllValuesAsync(user).then(callback);
     }
 
-    getAllValuesAsync(user?: User): Promise<SettingKeyValue[]> {
+    async getAllValuesAsync(user?: User): Promise<SettingKeyValue[]> {
         this.options.logger.debug("getAllValuesAsync() called.");
-        return new Promise(async (resolve) => {
-            const settings = await this.getSettingsAsync();
-            if (!settings) {
-                this.options.logger.error("config.json is not present, returning empty array");
-                resolve([]);
-                return;
-            }
-
-            const keys: string[] = Object.keys(settings);
-
-            let result: SettingKeyValue[] = [];
 
-            keys.forEach(key => {
-                result.push({
-                    settingKey: key,
-                    settingValue: this.evaluator.Evaluate(settings, key, void 0, user ?? this.defaultUser).Value
-                });
-            });
+        let result: SettingKeyValue[];
+        user ??= this.defaultUser;
+        try {
+            const [settings, remoteConfig] = await this.getSettingsAsync();
+            const [evaluationDetailsArray, errors] = evaluateAll(this.evaluator, settings, user, remoteConfig, this.options.logger);
+            if (errors?.length) {
+                throw typeof AggregateError !== "undefined" ? new AggregateError(errors) : errors.pop();
```

Comment:
AFAIK, not necessarily. `AggregateError` is a relatively new feature as it was introduced in ES2021:
* https://exploringjs.com/impatient-js/ch_new-javascript-features.html#new-in-es2021
* https://262.ecma-international.org/12.0/

Also, it's not polyfilled currently as I don't consider it critical but rather a minor enhancement in case the runtime supports it.